pkg/testing/define/define.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package define
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+	"github.com/elastic/go-elasticsearch/v8"
+
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/version"
+)
+
+// Require defines what this test requires for it to be run by the test runner.
+//
+// This must be defined as the first line of a test, otherwise the test runner.
+func Require(t *testing.T, req Requirements) *Info {
+	return defineAction(t, req)
+}
+
+type Info struct {
+	// ESClient is the elasticsearch client to communicate with elasticsearch.
+	// This is only present if you say a cloud is required in the `define.Require`.
+	ESClient *elasticsearch.Client
+
+	// KibanaClient is the kibana client to communicate with kibana.
+	// This is only present if you say a cloud is required in the `define.Require`.
+	KibanaClient *kibana.Client
+
+	// Namespace should be used for isolating data and actions per test.
+	//
+	// This is unique to each test and instance combination so a test that need to
+	// read/write data to a data stream in elasticsearch do not collide.
+	Namespace string
+}
+
+// Version returns the version of the Elastic Agent the tests should be using.
+func Version() string {
+	ver := os.Getenv("TEST_DEFINE_AGENT_VERSION")
+	if ver == "" {
+		return version.GetDefaultVersion()
+	}
+	return ver
+}
+
+// NewFixture returns a new Elastic Agent testing fixture.
+func NewFixture(t *testing.T, opts ...atesting.FixtureOpt) (*atesting.Fixture, error) {
+	buildsDir := os.Getenv("TEST_DEFINE_AGENT_BUILD_DIR")
+	if buildsDir == "" {
+		projectDir, err := findProjectRoot()
+		if err != nil {
+			return nil, err
+		}
+		buildsDir = filepath.Join(projectDir, "build", "distributions")
+	}
+	f := atesting.LocalFetcher(buildsDir)
+	opts = append(opts, atesting.WithFetcher(f), atesting.WithLogOutput())
+	return atesting.NewFixture(t, Version(), opts...)
```

Comment:
Correct me if I'm wrong, but the local fetcher would really only work with the local agent's default version (`version.GetDefaultVersion()`), right?  If a developer specified `TEST_DEFINE_AGENT_VERSION` as something other than the local agent's default version, the local fetcher wouldn't be able to find a locally-built Agent of that version, right?

Reply:
If that build of the Elastic Agent was present in the "TEST_DEFINE_AGENT_BUILD_DIR" or in the "${project_root}/build/distributions" it would work just fine.

Reply:
Okay, so basically if a developer wants to use the local fetcher with a different version than the local agent's default version, they most likely should specify `TEST_DEFINE_AGENT_VERSION` and `TEST_DEFINE_AGENT_BUILD_DIR` both.   It's unlikely that a build of the different version happens to exist in `${project_root}/build/distributions`.   I think we should document this as a recommendation or a tip.

Reply:
I was original thinking it was going to be used only by the runner and provide the user with a way to set those externally to that. But thinking about it more and now that the Fixture is also returned in local mode, it would make sense just to make these simpler environment variables. Like `AGENT_VERSION` and `AGENT_BUILD_DIR`. Thoughts?

Reply:
Yeah, I like the idea of simplifying the names since developers could set them.